src/coprocessor/statistics/cmsketch.rs

```diff
@@ -44,13 +46,32 @@ impl CmSketch {
         }
     }
 
+    pub fn sub(&mut self, bytes: &[u8], cnt: u32) {
+        self.count -= cnt;
+        let (h1, h2) = CmSketch::hash(bytes);
+        for (i, row) in self.table.iter_mut().enumerate() {
+            let j = (h1.wrapping_add(h2.wrapping_mul(i as u64)) % self.width as u64) as usize;
+            row[j] = row[j].saturating_sub(cnt);
+        }
+    }
+
+    pub fn push_to_top_n(&mut self, b: &[u8], cnt: u64) {
```

Comment:
Suggest changing function sig to `b: Vec<u8>` to let the caller know that an owned `Vec` is expected.